voluptuous/schema_builder.py

```diff
@@ -675,6 +677,46 @@ def _compile_list(self, schema):
         """
         return self._compile_sequence(schema, list)
 
+    def _compile_set(self, schema):
+        """Validate a set.
+
+        A set is an unordered collection of unique elements.
+
+        >>> validator = Schema({int})
+        >>> validator(set([42])) == set([42])
+        True
+        >>> with raises(er.Invalid, 'expected a set'):
+        ...   validator(42)
+        >>> with raises(er.MultipleInvalid, 'invalid value in set'):
+        ...   validator(set(['a']))
+        """
+        type_ = type(schema)
+        type_name = 'set' if type_ == set else 'frozenset'
```

Comment:
Can this be `type_name = type_.__name__`? That will also handle user-defined subclasses of set, etc.